src/screens/DiscoveryScreenV2/fragment/UsersFragment.js

```diff
@@ -109,14 +109,20 @@ const UsersFragment = ({
   }, []);
 
   React.useEffect(() => {
+    handleActiveSections();
+  }, [searchText, followedUsers]);
+
+  const handleActiveSections = () => {
+    if (!searchText) return;
+
     if (searchText.length === 0) {
       setActiveSections([]);
     } else if (searchText.length >= 0 && followedUsers.length > 0) {
       setActiveSections([0]);
     } else {
       setActiveSections([]);
     }
-  }, [searchText, followedUsers]);
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `handleActiveSections` function is defined after it's used in the `useEffect` hook. This could lead to a ReferenceError if `handleActiveSections` is not defined before the `useEffect` hook runs. To fix this, you should define `handleActiveSections` before the `useEffect` hook.

```diff
-  React.useEffect(() => {
-    handleActiveSections();
-  }, [searchText, followedUsers]);
-
-  const handleActiveSections = () => {
+  const handleActiveSections = () => {
     if (!searchText) return;
 
     if (searchText.length === 0) {
       setActiveSections([]);
     } else if (searchText.length >= 0 && followedUsers.length > 0) {
       setActiveSections([0]);
     } else {
       setActiveSections([]);
     }
  };

+  React.useEffect(() => {
+    handleActiveSections();
+  }, [searchText, followedUsers]);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->